utils/travis_runner.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+
+### BEGIN ###
+# Author: idevz
+# Since: 08:52:47 2019/07/08
+# Description:       travis_runner.sh
+# travis_runner          ./travis_runner.sh
+#
+# Environment variables that control this script:
+#
+### END ###
+
+set -ex
+
+export_or_prefix() {
+    if [ $TRAVIS_OS_NAME = osx ]; then
+        export OPENRESTY_PREFIX=$(TMP='./v_tmp' && $(which openresty) -V &>${TMP} &&
+            cat ${TMP} | grep prefix | grep -Eo 'prefix=(.*?)/nginx' |
+            grep -Eo '/.*/' && rm ${TMP})
+    else
+        export OPENRESTY_PREFIX="/usr/local/openresty-debug"
+    fi
+}
+
+before_install() {
+    sudo cpanm --notest Test::Nginx IPC::Run >build.log 2>&1 || (cat build.log && exit 1)
+}
+
+do_install() {
+    if [ $TRAVIS_OS_NAME = osx ]; then
```

Comment:
I think it's more convenient for different systems to use separate scripts.

how about: `***_mac.sh`, `***_linux.sh` ?

Reply:
if using like `***_mac.sh`, `***_linux.sh` things, the control logic about to using which one will need to write in .travis.yaml, I think it's so ugly.

Reply:
how about this way in travis: `bash utils/travis_$TRAVIS_OS_NAME.sh`